### DISCUSSION TITLE
How is the scroll shadow implemented?

### DISCUSSION THREAD

**Discussion by Toshinaki** [icflorescu/mantine-datatable#594] [Q&A]
Loving this datatable, especially the details.

When the table is scrollable, shadow appears and if scroll to right, shadow on the right disappears and shadow on the left show.
I checked the dev tools but found nothing.
I'm curious how is this implemented. Could you please point me to the related codes?

**Comment by icflorescu** [comment]
Hello Toshinaki, thank you for your kind words and apologies for the delayed reply.
The details you mentioned were very important for me when I started the project and I took pride in making it as "pleasing to the eye" as possible, hence the shadows.

It's CSS, mostly. The code is a bit complex, but here is the main idea: the table is wrapped inside a `ScrollArea` component that also includes four absolutely-positioned `div` elements, which have some CSS classes displaying gradients applied depending on the scroll status. 

The relevant parts should be [here](https://github.com/icflorescu/mantine-datatable/blob/main/package/DataTableScrollArea.tsx) and [here](https://github.com/icflorescu/mantine-datatable/blob/main/package/DataTableScrollArea.css). The gradient backgrounds are defined [here](https://github.com/icflorescu/mantine-datatable/blob/main/package/DataTable.css).
